src/01_getting_started/02_why_async.md

```diff
@@ -1,45 +1,121 @@
+Although details vary between different languages and frameworks,
+a brief overview of the main categories can help you understand how
+asynchronous programming fits within the field of concurrent programming:
+
+- **OS threads** don't require any changes to the programming model,
+  which makes them easier to
+  use than async. However, they come with a large performance overhead.
+  Thread pools can mitigate some of these costs, but not enough to support
+  massive IO-bound workloads.
+- **Event-driven programming**, in conjunction with _callbacks_, can be very
+  performant, but tends to result in a verbose, "non-linear" control flow.
+  Data flow and error propagation is often hard to follow.
+- **Coroutines**, like threads, don't require changes to the programming model,
+  which makes them easy to use. Like async, they can also support a large
+  number of tasks. However, they abstract away low-level details that
+  are important for systems programming and custom runtime implementors.
+- **The actor model** divides all concurrent computation into units called
+  actors, which communicate through fallible message passing, much like
+  in distributed systems. It can be efficiently implemented, but it leaves
+  many practical issues unanswered, such as flow control and retry logic.
+
+In short, asynchronous programming allows highly performant implementations
+that are suitable for low-level languages like Rust, while providing
+most of the ergonomic benefits of threads and coroutines.
+
+## Async in Rust vs other languages
+
+Although asynchronous programming is supported in many languages, some
+details vary across implementations. Rust's implementation of async
+differs from most languages in a few ways:
+
+- **Futures are inert** in Rust and make progress only when polled. Dropping a
+  future stops it from making further progress.
+- **Async is zero-cost** in Rust, and does not require heap allocations or
+  dynamic dispatch.
+  As such, you can use async Rust in constrained environments, such as embedded
+  systems.
+- **No built-in runtime** is provided by Rust. Instead, runtimes are provided by
+  community maintained crates.
+- **Both single- and multithreaded** runtimes are available in Rust, which have
+  different strengths and weaknesses.
+
+## Async vs threads in Rust
+
+The most obvious alternative to async in Rust is using OS threads, either
+directly through [`std::thread`](https://doc.rust-lang.org/std/thread/)
+or indirectly through a thread pool.
+Migrating from threads to async or vice versa
+typically requires major refactoring work, both in terms of implementation and
+(if you are building a library) any exposed public interfaces. As such,
+picking the model that suits your needs early can save you a lot of time.
+
+Note that asynchronous programming is not _better_ than threads, but different.
+If you don't need async for performance reasons, threads can often be
```

Comment:
This reads a bit confusingly; it sounds like you're about to talk about situations where OS threads are better, but instead it is more similar to the pros/cons comparison from above.

Reply:
Yes, good catch. Moved para down.